### ISSUE TITLE
Multiple GetCollection actions with different filters

### ISSUE THREAD

**Issue by vazgen** [api-platform/api-platform#2419]
Hello, I have an entity called books and I added 2 GetCollection actions. 
1. the first one is using regular ApiFilters
2. the second one is being used for book recommendations that does not need the default filters.

```
new GetCollection(
        normalizationContext: [
            'openapi_definition_name' => ModelInterface::LIST,
            'groups' => [...Book::GROUP_LIST],
            'skip_null_values' => false,
        ]),
    new GetCollection(
        uriTemplate: '/book-recommendation',
        controller: BookRecommendationAction::class,
        normalizationContext: [
            'openapi_definition_name' => ModelInterface::LIST,
            'groups' => [...Book::GROUP_LIST],
            'skip_null_values' => false,
        ],
        security: 'is_granted("ROLE_USER")',
    ),
```

Is it possible to remove ApiFilters for the second action?

**Comment by soyuka** [comment]
yes, declare your filters on the operation see https://api-platform.com/docs/core/filters/#doctrine-orm-and-mongodb-odm-filters

**Comment by mwierzbi** [comment]
If you want to have a separate filter per operation you must define all filters as service configuration. 
```services:
  book.date_filter:
    parent: 'api_platform.doctrine.orm.date_filter'
    arguments: [ { createdAt: ~ } ]
    tags:  [ 'api_platform.filter' ]
   ```
and add it to the operation
```
filters: ['book.date_filter']
```

If you use `ApiFilter` attribute and filters per operation all filters will be merged (https://github.com/api-platform/core/blob/db93dbc75fc4d0a45108fda874a840cc814dba8a/src/Metadata/Resource/Factory/FiltersResourceMetadataCollectionFactory.php#L54).

**Comment by Enz000** [comment]
> 

Thanks for this solution, there’s another way to do this without creating a yaml file?

Same situation, i have many ApiFilter declared on my resource, and i would like to enable only one for a specific operation

**Comment by EfrosIonelu** [comment]
@soyuka 
can you update ApiFilter constructor, to have a extra parameter operations: null|array<string>., 
if is null or empty to be applied on all collection operations
if is not empty to be applied only on operations from the list

for example: 
i have 2 collection operations for single entity 
one will have exact filtering on 2 fields because there are some fields that are filtered from the provider for example and allowing filtering by a field that will confuse people in api doc is not ok.
the second operations is a public one and has all possible filters

this is why we will need 

```
#[ApiFilter(SearchFilter::class, properties: ['key1' => 'partial', ''key2' => 'partial', 'operations': ['o1', 'o2']])]
#[ApiFilter(SearchFilter::class, properties: ['key3' => 'partial', 'operations': ['o2']])]
```

**Comment by soyuka** [comment]
Closing per #2987. Questions are best asked on [GitHub Discussions](https://github.com/api-platform/core/discussions) or [Stack Overflow (api-platform tag)](https://stackoverflow.com/questions/tagged/api-platform.com). The framework now lives at `api-platform/core`.
